mantidproject/mantid
# Allow Determination of Workspace Order from Output of PolarizationCorrectionWildes

### Description of work

#### Summary of work
Adds a new `SpinStates` property to the algorithm that allows the user to select the order that the spin state (`++`, `-+`, etc.) workspaces appear in the output WorkspaceGroup.


Fixes #35067 

#### Further detail of work
- Adds the new property to the algorithm, ensuring that they appear in the right order when it has been set. 
- Adds validation to ensure that the correct number of states is provided for a given correction.
- Amends the `SpinStateValidator` to handle SpinStates (`++`) and FlipperConfigurations (`00`). 
- Updates the documentation to account for the new functionality

### To test:
1. Run the test from [this documentation](https://docs.mantidproject.org/nightly/algorithms/PolarizationCorrectionWildes-v1.html#id12)
2. Take note of the order of the workspaces in the output WorkspaceGroup. 
3. Add a line under `Flippers='00, 11'` and add `SpinStates=`--,++,-+,+-`. 
4. Run the script again. 
5. Check the output workspace group is in the order you selected. 

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
